factory-cli/src/services/source/typescript/typescript-source.service.ts

```diff
@@ -1,22 +1,29 @@
 import * as _path from 'path';
-import { TypescriptSourceFactory } from "../../factories/source/typescript-source.factory";
-import { ISourceFactory } from "../../factories/source/source.factory";
-import { SolutionProject } from "../../models/solution";
-import { FileService, IFileService } from '../file.service';
+import { ISourceFactory } from "../../../factories/source/source.factory";
+import { TypescriptSourceFactory } from "../../../factories/source/typescript-source.factory";
+import { SolutionProject } from "../../../models/solution";
+import { FileService, IFileService } from '../../file.service';
+import { IAppConfiguration } from './app-configuration/app-configuration.interface';
+import { MysqlAppConfigurationService } from './app-configuration/mysql-app-configuration.service';
+import { SqliteAppConfigurationService } from './app-configuration/sqlite-app-configuration.service';
 
 export interface ITypescriptSourceService {
   createAppConfigFromSampleConfig: (solutionFolderPath: string, project: SolutionProject) => Promise<void>;
-  addMySqlAppConfigurationJson: (solutionFolderPath: string, project: SolutionProject) => Promise<void>;
-  addMySqlAppConfigurationModel: (solutionFolderPath: string, project: SolutionProject) => Promise<void>;
+  addAppConfigurationJson: (solutionFolderPath: string, project: SolutionProject, databaseType: string) => Promise<void>;
+  addAppConfigurationModel: (solutionFolderPath: string, project: SolutionProject, databaseType: string) => Promise<void>;
   addDataContextCompositionType: (solutionFolderPath: string, project: SolutionProject, contextName: string) => Promise<void>;
   addDataContextComposition: (solutionFolderPath: string, project: SolutionProject,
-    databaseProjectName: string, contextName: string) => Promise<void>;
+    databaseProjectName: string, contextName: string, configName: string) => Promise<void>;
 }
 
 export class TypescriptSourceService implements ITypescriptSourceService {
 
   fileService: IFileService = new FileService();
   sourceFactory: ISourceFactory = new TypescriptSourceFactory();
+  appConfigurationServiceFactory: IAppConfiguration[] = [
```

Comment:
This isn't a factory, just an array, so we should probably rename (for clarity).